Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts

```diff
@@ -8,25 +8,29 @@ import * as ci from './cieventlogger';
 import { exec } from 'child_process';
 const perf = require('performance-now');
 
-const executionStartTime = perf();
-const osPlat: string = os.platform();
 const packageName = 'Microsoft.TestPlatform';
 let packageSource = 'https://api.nuget.org/v3/index.json';
 
 async function startInstaller() {
-    tl.setResourcePath(path.join(__dirname, 'task.json'));
-    ci.publishEvent('Start', { OS: osPlat, isSupportedOS: (osPlat === 'win32').toString(), startTime: executionStartTime } );
-
-    if (osPlat !== 'win32') {
-        // Fail the task if os is not windows
-        tl.setResult(tl.TaskResult.Failed, tl.loc('OnlyWindowsOsSupported'));
-        return;
-    }
+    let executionStartTime;
 
     try {
+        const osPlat: string = os.platform();
+        executionStartTime = perf();
+
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+
         console.log(tl.loc('StartingInstaller'));
         console.log('==============================================================================');
```

Comment:
can this move up as the first one..? even before the call to os.platform ?

Reply:
So we need to load the resource file before i can shoot any localized messages to the console.

For the diagnostic version of the task we are going to upload i was just simply thinking of adding a tl.debug(i++); before every statement in the script.

Reply:
requires localization to output this